### DISCUSSION TITLE
How to add metronome to the waveplayer

### DISCUSSION THREAD

**Discussion by logesh2496** [katspaugh/wavesurfer.js#2960] [Q&A]
Hi, I'm looking to add metronome to the waveform player, I couldn't find any plugin so far. Thanks

**Comment by katspaugh** [comment]
You can write or use an existing metronome in Web Audio and start it when you start playing audio with wavesurfer. It doesn't have to be a plugin.

```js
const metronome = new Metronome(bpm)
const wavesurfer = new WaveSurfer({ ... })
wavesurfer.on('play', () => metronome.start())
wavesurfer.on('pause', () => metronome.stop())
```

See these links:
* https://github.com/cwilso/metronome
* https://blog.paul.cx/post/metronome/
